src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_AL_SIZE = 10;
+    private static final double CAPACITY_GROWTH_VALUE = 1.5;
+    private static final int ZERO_NUMBER = 0;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_AL_SIZE];
+        size = ZERO_NUMBER;
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add value by this index");
```

Comment:
The index check logic is repeated in multiple methods. Consider refactoring it into a separate private method to avoid code duplication.